src/common/command_buffers/details/clear.hpp

```diff
@@ -169,4 +169,147 @@ namespace commandbuffers
   public:
     int stencil;
   };
+
+  // Base class for ClearBuffer commands that use vector values
+  template <typename Derived, CommandBufferType Type, typename ValueType>
+  class ClearBufferBaseRequest : public TrCommandBufferSimpleRequest<Derived, Type>
+  {
+  public:
+    ClearBufferBaseRequest() = delete;
+    ClearBufferBaseRequest(int buffer, int drawbuffer, const std::vector<ValueType> &values)
+        : TrCommandBufferSimpleRequest<Derived, Type>()
+        , buffer(buffer)
+        , drawbuffer(drawbuffer)
+        , values(values)
+    {
+    }
+    ClearBufferBaseRequest(const ClearBufferBaseRequest &that, bool clone = false)
+        : TrCommandBufferSimpleRequest<Derived, Type>(that, clone)
+        , buffer(that.buffer)
+        , drawbuffer(that.drawbuffer)
+        , values(that.values)
```

Comment:
Copy constructor can't directly copy the STL values, see https://github.com/M-CreativeLab/jsar-runtime/blob/main/src/common/command_buffers/details/uniforms.hpp#L112-L119